src/ISA/ISA/ArcTypes/ArcTypes.fs

```diff
@@ -1683,6 +1683,28 @@ type ArcInvestigation(identifier : string, ?title : string, ?description : strin
             let copy = inv.Copy()
             copy.DeregisterMissingAssays()
             copy
+    
+    /// Updates the IOtypes of the IO columns across all tables in the investigation if possible.
+    ///
+    /// If an entity with the same name as an entity with a higher IOType specifity is found, the IOType of the entity with the lower IOType specificity is updated.
+    ///
+    /// E.g. RawDataFile is more specific than Source, but less specific than DerivedDataFile.
+    ///
+    /// E.g. Sample is equally specific to RawDataFile.
+    member this.UpdateIO() =
```

Comment:
And should the whole logic not be based on:

> Output with same id as Input overwrites input IOType

Because this would resolve deprecation?

Reply:
Renamed and added some documentation. I would keep it here, as it can also be used to check up on erroneous, non-deprecated arcs. And also the functionality needs to be spanning across files, making it infeasible to keep only in ISA IO.